python/doc/source/other_systems.rst

```diff
@@ -0,0 +1,116 @@
+.. software distributed under the License is distributed on an
+.. "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+.. KIND, either express or implied.  See the License for the
+.. specific language governing permissions and limitations
+.. under the License.
+
+.. currentmodule:: pyarrow
+.. _other_systems:
+
+Using Arrow with other systems
+==============================
+
+Timestamps
+----------
+
+Timestamps are data structures that mark a particular point in time
+and we want to be able to order them, regardless of where
+they originated.
+For human consumption timestamps are usually specified by the date
+together with the time of day, often using the local time zone.
+The problem with this scheme is that if things need to be ordered
+by time across multiple time zones, using local time can be ambiguous.
+Therefore timestamps from multiple time zones should always be collected,
+stored and communicated in UTC to avoid this ambiguity. 
+
+Timestamp types in Arrow are specified with a resolution and optional
+time zone. Several utility functions exist to convert data from 
+Pandas to Arrow, including functions that convert timestamp values 
+to milliseconds, because Pandas uses nanoseconds and other systems,
+i.e. Parquet use timestamps in milliseconds.
+
+In Arrow, timestamps have two forms, depending if the time zone is
+specified or not:
+
+*   **Time zone naive** (where ``tz=None`` in Python); there is no 
+    notion of UTC or local time zone.
+
+*   **Time zone aware** where the integer values are internally 
+    normalized to UTC.
+
+Apache Spark
+++++++++++++
+
+When using Arrow with `Apache Spark <http://spark.apache.org/>`_
+special attention must be given to conversion of timestamps.
+
+Spark, unlike Arrow, does treat timestamp data without a time zone
+as being in the local time zone of the server running the code.
+Spark version 2.2 allows for setting a configuration variable:
+``spark.sql.session.timeZone`` to ``UTC``. This configuration
+variable does only work on timestamps generated with ``SparkSQL``.
+In order to force timestamps coming from other sources, i.e.
+external files to be interpreted as UTC the following code must
+be executed before the operation::
+
+    os.environ["TZ"] = "UTC"
+    time.tzset()
+
+
+Also be careful when using the ``from_utc_timestamp`` function in Spark,
+as it returns a simple ``datetime`` object that does not contain any
+time zone information, and only shifts the time value based on the
+time zone and date specified. In the following case the shift is 6
+hours which is the offset Denver is behind UTC during 
+Daylight Saving Time::
+
+    In [9]: from pyspark.sql.functions import from_utc_timestamp
+
+    In [10]: df = sqlContext.createDataFrame([('2017-09-12 22:30:00',)], ['t'])
+
+    In [11]: df.select(from_utc_timestamp(df.t, "America/Denver").alias('t')).collect()
+    Out[11]: [Row(t=datetime.datetime(2017, 9, 12, 16, 30))]
```

Comment:
Maybe also show the numeric value of the timestamp to demonstrate the difference?

Reply:
Done!